pkg/shared/families/malware/yara/util/yara_util.go

```diff
@@ -0,0 +1,126 @@
+// Copyright © 2023 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/openclarity/vmclarity/pkg/shared/families/malware/common"
+)
+
+const (
+	// A yara detected malware line should be in the following format: <yara rule name> [<coma,separated,meta data>] <path>.
+	scanError                   = "error scanning"
+	yaraRuleNamePosition        = 0
+	yaraRuleMetadataPosition    = 1
+	detectedMalwarePathPosition = 2
+	malwareNameString           = "name" // The key defined in the metadata of yara rule
+	malwareTypeString           = "type" // The key defined in the metadata of yara rule
+	yaraOutputFiledNum          = 3      // Number of yara output fields. <yara rule name> [<coma,separated,meta data>] <path>
+)
+
+func ParseYaraScanOutput(yaraOutput string, logger *logrus.Entry) []common.DetectedMalware {
```

Comment:
You are right. I think we need to update clam output parsing as well. So it can be a follow-up task for both of the yara and clam. We have to create a new function that has two args (command, parser_func) that runs the command and parse output line-by-line. WDYT?

Reply:
Sure, lets make sure we open a ticket to track it.